sw/device/silicon_creator/manuf/lib/provisioning.c

```diff
@@ -90,6 +88,95 @@ static_assert(OTP_CTRL_PARAM_CREATOR_ROOT_KEY_SHARE0_SIZE ==
                   OTP_CTRL_PARAM_CREATOR_ROOT_KEY_SHARE1_SIZE,
               "Detected Root key share size mismatch");
 
+// ECC curve to use with ECDH keygen.
+static const ecc_curve_t kCurveP256 = {
+    .curve_type = kEccCurveTypeNistP256,
+    .domain_parameter = NULL,
+};
+
+// ECDH private key configuration.
+static const crypto_key_config_t kEcdhPrivateKeyConfig = {
+    .version = kCryptoLibVersion1,
+    .key_mode = kKeyModeEcdh,
+    .key_length = kP256ScalarBytes,
+    .hw_backed = kHardenedBoolFalse,
+    .diversification_hw_backed = NULL,
+    .security_level = kSecurityLevelLow,
+};
+
+// ECDH shared secret configuration.
+static const crypto_key_config_t kEcdhSharedKeyConfig = {
+    .version = kCryptoLibVersion1,
+    .key_mode = kKeyModeAesGcm,
```

Comment:
As we discussed offline, a non-authenticated AES mode should suffice as we are working on #17496 to provide a simple (albeit non cryptographic) integrity check over the whole data payload sent from the device (not just the RMA token). Switched to ECB since RMA token is only 128 bits, and added a static assert to detect if the token size changes and we need to switch modes.